tools/flaky_test_monitor/level1/process_summary1_results_test.go

```diff
@@ -101,58 +154,24 @@ func runProcessSummary1TestRun(t *testing.T, jsonExpectedActualFile string) {
-	require.Equal(t, expectedTestRun.CommitDate, actualTestRun.CommitDate)
-	require.Equal(t, expectedTestRun.CommitSha, actualTestRun.CommitSha)
-	require.Equal(t, expectedTestRun.JobRunDate, actualTestRun.JobRunDate)
-	require.Equal(t, len(expectedTestRun.PackageResults), len(actualTestRun.PackageResults))
-
-	// check each package
-	for packageIndex := range expectedTestRun.PackageResults {
-		expectedPackageResults := expectedTestRun.PackageResults[packageIndex]
-		actualPackageResults := actualTestRun.PackageResults[packageIndex]
-
-		require.Equal(t, expectedPackageResults.Elapsed, actualPackageResults.Elapsed)
-		require.Equal(t, expectedPackageResults.Package, actualPackageResults.Package)
-		require.Equal(t, expectedPackageResults.Result, actualPackageResults.Result)
-		require.Empty(t, expectedPackageResults.TestMap, actualPackageResults.TestMap)
-
-		// check outputs of each package result
-		require.Equal(t, len(expectedPackageResults.Output), len(actualPackageResults.Output))
-		for packageOutputIndex := range expectedPackageResults.Output {
-			require.Equal(t, expectedPackageResults.Output[packageOutputIndex], actualPackageResults.Output[packageOutputIndex])
-		}
-
-		// check tests results of each package
-		checkTestResults(t, expectedPackageResults.Tests, actualPackageResults.Tests)
-	}
-	// finally, compare the entire actual test run against what's expected - if there were any discrepancies they should have been caught by now
-	require.Equal(t, expectedTestRun, actualTestRun)
-}
-
-func checkTestResults(t *testing.T, expectedTestResults []common.TestResult, actualTestResults []common.TestResult) {
-	require.Equal(t, len(expectedTestResults), len(actualTestResults))
-	for testResultIndex := range expectedTestResults {
-
-		expectedTestResult := expectedTestResults[testResultIndex]
-		actualTestResult := actualTestResults[testResultIndex]
+func checkLevel1Summaries(t *testing.T, expectedLevel1Summary common.Level1Summary, actualLevel1Summary common.Level1Summary) {
+	// number of results should be the same between expected and actual
+	require.Equal(t, len(expectedLevel1Summary.Rows), len(actualLevel1Summary.Rows))
 
-		// check all outputs of each test result
-		require.Equal(t, len(expectedTestResult.Output), len(actualTestResult.Output), fmt.Sprintf("TestResult[%d].Test: %s", testResultIndex, actualTestResult.Test))
-		for testResultOutputIndex := range expectedTestResult.Output {
-			require.Equal(t, expectedTestResult.Output[testResultOutputIndex], actualTestResult.Output[testResultOutputIndex], fmt.Sprintf("TestResult[%d] Output[%d]", testResultIndex, testResultOutputIndex))
-		}
+	// check that all expected results are in actual results
+	for actualRowIndex := range actualLevel1Summary.Rows {
+		require.Contains(t, actualLevel1Summary.Rows, expectedLevel1Summary.Rows[actualRowIndex], printLevel1TestResult(expectedLevel1Summary.Rows[actualRowIndex].TestResult, "expected not in actual"))
```

Comment:
This checks that the actual result contains the expected as a substring, but do we actually want to check that they are exactly equal?

Reply:
It checks by the entire element, not by substring. It checks that every expected level 1 result row row element (not a substring) is in the actual result list (and vice versa). This makes it more adaptable to expected or actual results being in a different order and still passing.

[`require.contains()`](https://pkg.go.dev/github.com/stretchr/testify/require#Contains) uses "asserts that the specified string, list(array, slice...) or map contains the specified substring or element"

Reply:
So, what you actually want to do here is check that `actualLevel1Summary.Rows` and `expectedLevel1Summary.Rows` contain the exact same elements right? Should we use something like `assert.EqualValues` here?

Reply:
I was hoping this would work, but it doesn't. I tried this:

`require.EqualValues(t, expectedLevel1Summary.Rows, actualLevel1Summary.Rows)`

but that causes many assertion failures.

> So, what you actually want to do here is check that actualLevel1Summary.Rows and expectedLevel1Summary.Rows contain the exact same elements right?

Yes, that's right. I used to compare each element one at a time between expected and actual lists but that was brittle because the order mattered. I found using `require.contains()` is a good way to do assertions without requiring the order is the same.

Reply:
https://pkg.go.dev/github.com/stretchr/testify/assert#ElementsMatch

Reply:
That works - thanks! Updated to use `require.ElementsMatch()`

https://github.com/onflow/flow-go/pull/2102/commits/4275b7c8535e8b65e1ee92d636efe2b93e320b4e